File: demo/src/main/java/uk/co/threefi/dataload/structure/CsvColumnifier.java
package uk.co.threefi.dataload.structure;

import java.io.Serializable;

public class CsvColumnifier implements Columnifier, Serializable {

  private static final long serialVersionUID = -812004521983071103L;

  private String delimiter=",";

  public CsvColumnifier(String delimiter) {
    this.delimiter = delimiter;
  }
  public String[] toColumns(String raw) {
    return raw.split(delimiter);
  }
}

File: demo/src/main/scala/kaspar/dataload/structure/CsvColumnifier.scala
package kaspar.dataload.structure

import kafka.serializer.{Decoder, StringDecoder}
import kafka.utils.VerifiableProperties
import org.apache.kafka.common.record.Record
import org.apache.kafka.common.utils.Utils

class CsvColumnifier(val delimiter: String = ",") extends Columnifier with Serializable {

  @transient var decoder: Decoder[String] = null

  override def toColumns(record: Record): scala.Seq[Any] = {

    if(decoder == null) {
      decoder = new StringDecoder(new VerifiableProperties)
    }

    val rawValue: String = record.offset + "," + record.timestamp + "," +
      decoder.fromBytes(Utils.readBytes(record.value))

    rawValue.split(delimiter)
  }
}
